extensions/azurecore/src/account-provider/azureAccountProvider.ts

```diff
@@ -95,13 +104,35 @@ export class AzureAccountProvider implements azdata.AccountProvider, vscode.Disp
 	private async _initialize(storedAccounts: AzureAccount[]): Promise<AzureAccount[]> {
 		const accounts: AzureAccount[] = [];
 		console.log(`Initializing stored accounts ${JSON.stringify(accounts)}`);
-		for (let account of storedAccounts) {
-			const azureAuth = this.getAuthMethod(account);
-			if (!azureAuth) {
-				account.isStale = true;
-				accounts.push(account);
+		const authLibrary = vscode.workspace.getConfiguration('azure').get('authenticationLibrary');
+		const updatedAccounts = storedAccounts.filter(account => {
+			if (account.key.authLibrary) {
+				if (account.key.authLibrary === authLibrary) {
```

Comment:
the logic of the method is hard to understand for me.  I'm reading as:

if the account key authLibrary is defined and current auth library is the same then we update, otherwise if it's ADAL we update.  What's the rationale for updating this subset of accounts?

Reply:
This is to handle the case where the account.key.authLibrary is not present, which will be the case if the account was added prior to the addition of this PR.  Since these accounts will have been added with ADAL, they will be treated as such.